esphome/components/nextion/nextion.cpp

```diff
@@ -1034,8 +1057,41 @@ void Nextion::add_no_result_to_queue_with_command_(const std::string &variable_n
 
   if (this->send_command_(command)) {
     this->add_no_result_to_queue_(variable_name);
+#ifdef USE_NEXTION_COMMAND_SPACING
+  } else {
+    // Command blocked by spacing, add to queue WITH the command for retry
+    this->add_no_result_to_queue_with_pending_command_(variable_name, command);
+#endif  // USE_NEXTION_COMMAND_SPACING
+  }
+}
+
+#ifdef USE_NEXTION_COMMAND_SPACING
+void Nextion::add_no_result_to_queue_with_pending_command_(const std::string &variable_name,
+                                                           const std::string &command) {
+#ifdef USE_NEXTION_MAX_QUEUE_SIZE
+  if (this->max_queue_size_ > 0 && this->nextion_queue_.size() >= this->max_queue_size_) {
+    ESP_LOGW(TAG, "Queue full (%zu), drop: %s", this->nextion_queue_.size(), variable_name.c_str());
+    return;
   }
+#endif
+
+  RAMAllocator<nextion::NextionQueue> allocator;
+  nextion::NextionQueue *nextion_queue = allocator.allocate(1);
+  if (nextion_queue == nullptr) {
+    ESP_LOGW(TAG, "Queue alloc failed");
+    return;
+  }
+  new (nextion_queue) nextion::NextionQueue();
+
+  nextion_queue->component = new nextion::NextionComponentBase;
+  nextion_queue->component->set_variable_name(variable_name);
+  nextion_queue->queue_time = millis();
```

Comment:
Any chance this can use the cached time? See #9104

Reply:
I would say it should be safe to use here

```suggestion
  nextion_queue->queue_time = App.get_loop_component_start_time();
```

Reply:
There are many other places on this component using `millis()`, so I would propose we take this in another PR.
I can commit to work on this asap.

Reply:
There is:
- #9150

Reply:
Great! I'm fine with deferring the change to #9150 -- thanks! 🍻